httpserver/http_connection.cpp

```diff
@@ -186,10 +188,20 @@ connection_t::connection_t(boost::asio::io_context& ioc, ssl::context& ssl_ctx,
       security_(security) {
 
     LOKI_LOG(trace, "connection_t");
+
     start_timestamp_ = std::chrono::steady_clock::now();
 }
 
-connection_t::~connection_t() { LOKI_LOG(trace, "~connection_t"); }
+connection_t::~connection_t() {
+
+    // Safety net
+    if (stream_.lowest_layer().is_open()) {
+        LOKI_LOG(warn, "Client socket should be closed by this point, but wasn't. Closing now.");
+        stream_.lowest_layer().close();
```

Comment:
Should we maybe move this `is_open` check to `do_close`, and just call `do_close` here?

Reply:
Will do in the upcoming PR!